### ISSUE TITLE
Plot margins

### ISSUE THREAD

**Issue by cpsievert** [ropensci/plotly#211]
As it appears on page 60 of [Hadley's PhD thesis](http://had.co.nz/thesis/practical-tools-hadley-wickham.pdf), there are perceptual issues with plotting data next to the margin in the plot. It goes on to say:

> For continuous data, 5% extra space is added, and for discrete data half a bar is added. These can be fine tuned by the user if necessary.

In plotly, we have the luxury of zooming/panning, but it'd be nice if we could match this default behaviour in ggplot2.

**Comment by cpsievert** [comment]
Whoops, I think this is related to the checkmark titled "add some more spacing around the graph: expand the range" [here](https://github.com/ropensci/plotly/issues/161).

**Comment by cpsievert** [comment]
For a plot with a single panel,  [layout(margin = ...)](https://plot.ly/r/reference/#layout) would be a solution, but what about multiple panels?

**Comment by cpsievert** [comment]
And, more generally, we should be respecting the expand argument:

```r
qplot(1:10, 1:10)
# versus
qplot(1:10, 1:10) + 
  scale_x_continuous(expand = c(0,0)) + 
  scale_y_continuous(expand = c(0,0))
```

**Comment by westonplatter** [comment]
@cpsievert happy to help out on this since I ran into this issue - any pointers you might be able to provide before I jump in?

**Comment by cpsievert** [comment]
The "proper" way to do this is probably going to change when the new version of ggplot2 hits. So, if you want to tackle this, I recommend working from [our fix for the dev version of ggplot2](https://github.com/ropensci/plotly/pull/269). This will likely be merged within a month from now when the new ggplot2 hits CRAN.

Right [around here](https://github.com/ropensci/plotly/blob/164a6ad58c7d6261ad89de524e64214b20812968/R/ggplotly.R#L429) we start to translate axis information. Somewhere in there we need to set the proper range for each axis. Problem is, I'm not exactly sure the best way to find the range from the plot object. Once we have the proper ranges, it won't be hard to translate to plotly: 

```r
p <- qplot(data = mtcars, mpg, geom = "density")
g <- plotly_build(p)
g$layout$xaxis <- modifyList(g$layout$xaxis, list(range = c(5, 50)))
g
```

**Comment by cpsievert** [comment]
This should be fixed in f1900cc3cd464408a60ffadfe3e95460747430e0.
